sw/host/tests/rom/e2e_bootstrap_rma/src/main.rs

```diff
@@ -18,362 +21,315 @@ use opentitanlib::dif::lc_ctrl::{
-            "-c",
-            "jtag newtap lc_ctrl tap -irlen 5 -expected-id 0x04f5484d -ignore-bypass;",
-            "-c",
-            &format!(
-                "target create {JTAG_TAP_NAME} riscv -chain-position lc_ctrl.tap -rtos hwthread;"
-            ),
-            "-c",
-            "init;",
-        ])
-        .args(["-f", SCRIPT_FILE_PATH])
-        .args(["-c", "shutdown;"])
-        .stdout(Stdio::inherit())
-        .stderr(Stdio::inherit())
-        .spawn()?
-        .wait()?;
-
-    if !openocd_exit.success() {
-        bail!("OpenOCD exited with status {}", openocd_exit);
+    // Read back the values to confirm the write.
+    for (reg, value) in iter::zip(token_regs, token_words) {
+        assert_eq!(jtag.read_lc_ctrl_reg(reg)?, value);
     }
 
-    // Reset the device and confirm that its lifecycle state is now RMA by
-    // reading the "LCV:" line from the UART.
-    let uart = transport.uart("console")?;
-    reset(transport, &[], opts.init.bootstrap.options.reset_delay)?;
-    let mut console = UartConsole {
-        timeout: Some(CONSOLE_TIMEOUT),
-        exit_success: Some(Regex::new(&format!(
-            "LCV:{:x}\r\n",
-            DifLcCtrlState::Rma.redundant_encoding()
-        ))?),
-        exit_failure: Some(Regex::new(r"LCV:[0-9a-f]+\r\n")?),
-        ..Default::default()
-    };
-    let result = console.interact(&*uart, None, Some(&mut std::io::stdout()))?;
-    if result != ExitStatus::ExitSuccess {
-        bail!("FAIL: {:?}", result);
-    };
-    Ok(())
-}
+    // Set the transition target to `RMA` and read back to confirm.
+    jtag.write_lc_ctrl_reg(
+        &LcCtrlReg::TransitionTarget,
+        DifLcCtrlState::Rma.redundant_encoding(),
+    )?;
+    assert_eq!(
+        jtag.read_lc_ctrl_reg(&LcCtrlReg::TransitionTarget)?,
+        DifLcCtrlState::Rma.redundant_encoding(),
+    );
 
-/// Orchestrates iteration 1 of the rom_bootstrap_rma testpoint. This never
-/// issues the life cycle RMA command. It verifies that the ROM times out on
-/// spin cycles, automatically resets the device, and logs the expected reset
-/// reasons.
-fn test_no_rma_command(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
-    let uart = transport.uart("console")?;
-    reset(
-        transport,
-        &["RMA_BOOTSTRAP"],
-        opts.init.bootstrap.options.reset_delay,
+    log::info!("Initiating lifecycle transition");
+
+    // Write `START` to the `TRANSITION_CMD` register.
+    jtag.write_lc_ctrl_reg(
+        &LcCtrlReg::TransitionCmd,
+        LcCtrlTransitionCmdBit::Start.mask(),
     )?;
 
-    log::info!("Waiting for reset reasons on console...");
+    // NOTE: These two checks race against the chip resetting. Can we halt the
+    // clock while these are being checked?
+    assert_eq!(jtag.read_lc_ctrl_reg(&LcCtrlReg::TransitionRegwen)?, 0);
+    assert_eq!(
+        jtag.read_lc_ctrl_reg(&LcCtrlReg::Status)?,
+        LcCtrlStatusBit::Initialized.mask()
+    );
```

Comment:
Thinking about this more, `TRANSITION_REGWEN` should always be set to `0` here before or after the reset. Not sure about the `INITIALIZED` status. In theory, I think we could hit this in the tiny gap where the LC isn't initialized yet, but that seems unlikely.